modules/framework/galasa-parent/dev.galasa.framework.api.users/src/test/java/dev/galasa/framework/api/users/internal/routes/UsersDeleteRouteTest.java

```diff
@@ -0,0 +1,180 @@
+
+
+    GalasaGson gson = new GalasaGson(); 
+    Map<String, String> headerMap = Map.of("Authorization", "Bearer " + BaseServletTest.DUMMY_JWT);
+
+    @Test
+    public void testUsersDeleteRequestReturnsNotFoundDueToMissingUserDoc() throws Exception {
+        // Given...
+        MockEnvironment env = new MockEnvironment();
+        MockTimeService mockTimeService = new MockTimeService(Instant.now());
+        MockAuthStoreService authStoreService = new MockAuthStoreService(mockTimeService);
+        MockFramework framework = new MockFramework(authStoreService);
+
+        String baseUrl = "http://my.server/api";
+
+        String loginId = "admin";
+
+        env.setenv(EnvironmentVariables.GALASA_USERNAME_CLAIMS, "preferred_username");
+        env.setenv(EnvironmentVariables.GALASA_EXTERNAL_API_URL,baseUrl);
+        MockUsersServlet servlet = new MockUsersServlet(framework, env);
+
+        MockHttpServletRequest mockRequest = new MockHttpServletRequest("/" + loginId, headerMap);
+        mockRequest.setMethod(HttpMethod.DELETE.toString());
+
+        MockHttpServletResponse servletResponse = new MockHttpServletResponse();
+        ServletOutputStream outStream = servletResponse.getOutputStream();      
+
+        // When...
+        servlet.init();
+        servlet.doDelete(mockRequest, servletResponse);
+
+        assertThat(servletResponse.getStatus()).isEqualTo(404);
+        checkErrorStructure(outStream.toString(), 5083, "GAL5083E",
+            "Unable to retrieve a user with the given ‘loginId’. No such user exists. Check your request query parameters and try again.");
+        assertThat(servletResponse.getContentType()).isEqualTo("application/json");
+    }
+
+    @Test
+    public void testUsersDeletesAUserReturnsOK() throws Exception {
+        // Given...
+        MockEnvironment env = new MockEnvironment();
+        MockTimeService mockTimeService = new MockTimeService(Instant.now());
+        MockAuthStoreService authStoreService = new MockAuthStoreService(mockTimeService);
+        MockFramework framework = new MockFramework(authStoreService);
+
+        String baseUrl = "http://my.server/api";
+
+        String loginId = "user-1";
+
+        env.setenv(EnvironmentVariables.GALASA_USERNAME_CLAIMS, "preferred_username");
+        env.setenv(EnvironmentVariables.GALASA_EXTERNAL_API_URL,baseUrl);
+        MockUsersServlet servlet = new MockUsersServlet(framework, env);
+
+        MockHttpServletRequest mockRequest = new MockHttpServletRequest("/" + loginId, headerMap);
+        mockRequest.setMethod(HttpMethod.DELETE.toString());
+
+        MockHttpServletResponse servletResponse = new MockHttpServletResponse();     
+        
+        MockUser mockUser1 = createMockUser("user-1", "docid", "web-ui");
+        authStoreService.addUser(mockUser1);
+
+        // When...
+        servlet.init();
+        servlet.doDelete(mockRequest, servletResponse);
+
+        assertThat(servletResponse.getStatus()).isEqualTo(204);
```

Comment:
We could perhaps assert that the user has been deleted from the mock auth store?